source/models/database.py

```diff
@@ -0,0 +1,55 @@
+import os, json
+
+from commands_functions.schema_keys import Keys
+from models.table import Table
+from outputs.exceptions import *
+
+class Database:
+    def __init__(self, DB_name = None, schema_path = None):
+        if schema_path is not None:
+            self.schema_data = self.validate_schema_path(schema_path)
+        else:
+            self.tables = {}
+            for table_schema in self.schema_data[Keys.TABLES]:
+                self.tables[table_schema[Keys.NAME]] = Table(self, table_schema)
```

Comment:
You should create the table map in case of passing schema as well

Reply:
done